interfaces/builtin/netlink_protocol.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/snap"
+)
+
+const netlinkProtocolSummary = `allows communication through the kernel custom netlink protocol`
+
+const netlinkProtocolBaseDeclarationSlots = `
+  netlink-protocol:
+    allow-installation:
+      slot-snap-type:
+        - core
+        - gadget
+    deny-auto-connection: true
+`
+
+// netlinkProtocolInterface type
+type netlinkProtocolInterface struct {
+	commonInterface
+}
+
+// BeforePrepareSlot checks the slot definition is valid
+func (iface *netlinkProtocolInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
+	// Must have a protocol number
+	number, ok := slot.Attrs["protocol"]
```

Comment:
this should be maybe called  custom-protocol

Reply:
Does it always have to custom?
Do we have interfaces for all the existing 23 protocols?
I thought this could be good way also to cover situations when we have no interface for existing 23 protocols

Reply:
netlink(7) calls this family,  so I think we want the following mandatory attributes:

```
slot side:
  family-name: a string of the alnum-alnum-* form  with at least length 2, the name of the driver/protocol
  family: integer

plug side:
  family-name
```

the idea of family-name is that it makes clearer whether a plug and slot are meant to be used together, also it can be used in snap declaration rules

Reply:
@alexmurray it's probably too awkward to enforce this here, but maybe review-tools can have a check emitting a warning if family is within the predefined range (0-22?), instead of outside

Reply:
I renamed the attribute as per @pedronis' suggestion